### PR TITLE
migrate run_all_tests to generic ct_run_tests_from_testcase

### PR DESCRIPTION
**Pull request by zmiklank**

### CODE REVIEW

**Comment by zmiklank** [comment]
[test-all]

**Comment by zmiklank** [comment]
[test-all]

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
[test-all]

File: test/run [hunk 1]
```diff
@@ -429,43 +398,34 @@ function run_dockerfiles_test() {
...
+TEST_SET=${TEST_LIST_PERL_APP} ct_run_tests_from_testset "perl-test-app"
 
 # Run the rest of the tests by defining TEST_LIST to include the rest of the relevant cases
-TEST_LIST=run_dockerfiles_test
+TEST_SET=run_dockerfiles_test
 
 # These tests only make sense for rhscl
 if [ "$OS" == "rhel7" ] || [ "$OS" == "centos7" ]; then
-  # autocleanup only enabled here as only the following tests so far use it
-  ct_enable_cleanup
```
Comment (phracek) [inline test/run:455]: Why this was removed?

File: test/run [hunk 2]
```diff
@@ -429,43 +398,34 @@ function run_dockerfiles_test() {
...
-echo "$test_short_summary"
-
-if [ $TESTSUITE_RESULT -eq 0 ] ; then
-  echo "Tests for ${IMAGE_NAME} succeeded."
-else
-  echo "Tests for ${IMAGE_NAME} failed."
-fi
-
-exit $TESTSUITE_RESULT
+ct_run_tests_from_testset "rest-of-the-tests"
```
Comment (phracek) [inline test/run:430]: What is the name 'rest-of-the-tests' used for?

I guess, 'run_dockerfiles_test' and 'test_scl_enable' can be used as TEST_REST_LIST variable or name it like 'dockerfiles-scl-enable-tests'

**Review by phracek — CHANGES_REQUESTED** [review]
Only two things. The rest looks good to me.

File: test/run (hunk 1 above)
Comment (zmiklank) [reply test/run:455]: Because it is added outside of this `if`.

File: test/run (hunk 2 above)
Comment (zmiklank) [reply test/run]: addressed in new commit.

**Comment by zmiklank** [comment]
[test-all]

**Comment by phracek** [comment]
RHEL9 failed for the reason:

```bash
Cloning into 'nginx-container'...
fatal: unable to access 'https://github.com/sclorg/nginx-container.git/': Could not resolve host: github.com
ERROR: Git repository https://github.com/sclorg/nginx-container.git cannot be cloned into nginx-container.
Terminating the Dockerfile build.
```

Let's try it one more time
[test]

**Review by phracek — APPROVED** [review]
LGTM. Thanks for addressing reviews. Let's get merge it as soon as tests are green except CentOS 7

**Pull request merged**